File: packages/types/src/config/presets.ts
/**
 * Preset Configuration Types
 * Unified preset configuration types for enabling/disabling predefined resources
 */

/**
 * Context Compression Preset Configuration
 */
export interface ContextCompressionPresetConfig {
  enabled: boolean;
  prompt?: string;
  timeout?: number;
  maxTriggers?: number;
}

/**
 * Predefined Tools Preset Configuration
 */
export interface PredefinedToolsPresetConfig {
  enabled: boolean;
  allowList?: string[];
  blockList?: string[];
  config?: {
    readFile?: {
      workspaceDir?: string;
      maxFileSize?: number;
    };
    writeFile?: {
      workspaceDir?: string;
    };
    editFile?: {
      workspaceDir?: string;
    };
    bash?: {
      defaultTimeout?: number;
      maxTimeout?: number;
    };
    sessionNote?: {
      workspaceDir?: string;
      memoryFile?: string;
    };
    backgroundShell?: {
      workspaceDir?: string;
    };
  };
}

/**
 * Predefined Prompts Preset Configuration
 */
export interface PredefinedPromptsPresetConfig {
  enabled: boolean;
}

/**
 * Presets Configuration
 */
export interface PresetsConfig {
  contextCompression?: ContextCompressionPresetConfig;
  predefinedTools?: PredefinedToolsPresetConfig;
  predefinedPrompts?: PredefinedPromptsPresetConfig;
}

File: crates/foundation/wf-types/src/config/presets.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
pub struct PredefinedToolsPresetConfig {
    pub enabled: Option<bool>,
    pub tools: Option<Vec<String>>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
pub struct PredefinedPromptsPresetConfig {
    pub enabled: Option<bool>,
    pub prompts: Option<Vec<String>>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
pub struct PresetsConfig {
    pub predefined_tools: Option<PredefinedToolsPresetConfig>,
    pub predefined_prompts: Option<PredefinedPromptsPresetConfig>,
}
